db_checker.py

```diff
@@ -37,122 +39,128 @@
+        for rec in self.db.records:
+            if rec.is_interest() and not rec.is_disable() and (rec.get_type() in EGU_sub_list):
+                unit = rec.get_field("EGU")
 
-                    if unit is None:
-                        err += 1
-                        print "ERROR: Missing units on " + err_src_fmt(db, rec)
+                if unit is None:
+                    err += 1
+                    failure_message += "   -> Missing units on {}\n".format(rec)
 
-        self.assertEqual(err, 0, msg="Missing units on interesting PVs in project")
+        self.assertEqual(err, 0, msg=failure_message)
 
+    @ignore(["DbUnitChecker"], "DB unit checker contains tests that should fail.")
     def test_interest_calc_readonly(self):
         """
         This method checks that interesting PVs that are calc fields are set to
         readonly
         """
         err = 0
+        failure_message = "Writable calc records in {}\n".format(self.db.directory)
 
-        for db in dbs:
-            for rec in db.records:
-                if rec.is_interest() and (rec.get_type() in ASG_list):
-                    value = rec.get_field("ASG")
+        for rec in self.db.records:
+            if rec.is_interest() and (rec.get_type() in ASG_list):
+                value = rec.get_field("ASG")
 
-                    if value != "READONLY":
-                        err += 1
-                        print "ERROR: Missing ASG on " + err_src_fmt(db, rec)
+                if value != "READONLY":
+                    err += 1
+                    failure_message += " Missing ASG on {}\n".format(rec)
 
-        self.assertEqual(err, 0, msg="Missing ASG on interesting calculation \
-                records in project")
+        self.assertEqual(err, 0, msg=failure_message)
 
+    @ignore(["DbUnitChecker"], "DB unit checker contains tests that should fail.")
     def test_desc_length(self):
         """
         This method checks that the description length on all PVs is no longer than 40 chars
         """
         err = 0
-        for db in dbs:
-            for rec in db.records:
-                desc = rec.get_field("DESC")
+        failure_message = "Description too long in {}\n".format(self.db.directory)
 
-                if desc is not None:
-                    # remove macros
-                    desc = re.sub(r'\$\([^)]*\)', '', desc)
+        for rec in self.db.records:
+            desc = rec.get_field("DESC")
 
-                    if len(desc) > 40:
-                        err += 1
-                        print "ERROR: Description too long on " + err_src_fmt(db, rec)
+            if desc is not None:
+                # remove macros
+                desc = re.sub(r'\$\([^)]*\)', '', desc)
 
-        self.assertEqual(err, 0, msg="Overly long description in project")
+                if len(desc) > 40:
+                    err += 1
+                    failure_message += "   -> Description too long on {}\n".format(rec)
```

Comment:
I gave some thought to whether this is the best way of combining the error messages. I don't think it's critical. `+=` is a lossy algorithm. It's quite often to see string concatenation prematurely leading to it having to be unpicked later.

In this case, when I get to the end of the test, I don't have all the details of what has failed, so you have to have an extra variable `err` to act as a counter.  You could instead have a collection of failure strings and then count the length of it. You can then separate your failure tracking and formatting concerns by doing a  `"\n".join(failures)` at the end. It might need a tad more memory, but not enough to worry about.

Perhaps you already considered this. It's not a major concern, just food for thought.